### PR TITLE
Skepticleo trainer argparser

### PR DESCRIPTION
**Pull request by williamFalcon**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Duplicate to #952  

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -822,6 +828,29 @@ def slurm_job_id(self) -> int:
...
         return job_id
 
+    @property
+    @classmethod
+    def default_attributes(cls):
+        return vars(cls())
+
+    @classmethod
+    def add_argparse_args(cls, parent_parser: ArgumentParser) -> ArgumentParser:
+
```
Comment (Borda) [inline pytorch_lightning/trainer/trainer.py]:
```suggestion
        """Extend existing argparse by default `Trainer` attributes."""
```

File: pytorch_lightning/trainer/trainer.py [hunk 2]
```diff
@@ -822,6 +828,29 @@ def slurm_job_id(self) -> int:
...
+        trainer_default_params = Trainer.default_attributes
+
+        for arg in trainer_default_params:
+            parser.add_argument('--{0}'.format(arg), default=trainer_default_params[arg], dest=arg)
+
+        return parser
+
+    @classmethod
+    def from_argparse_args(cls, args) -> Trainer:
```
Comment (Borda) [inline pytorch_lightning/trainer/trainer.py]:
```suggestion
    def from_argparse_args(cls, args: Namespace) -> Trainer:
```

File: tests/trainer/test_trainer.py [hunk 3]
```diff
@@ -600,3 +601,141 @@ def test_end(self, outputs):
 
     model = LightningTestModel(hparams)
     Trainer().test(model)
+
+@mock.patch('argparse.ArgumentParser.parse_args',
+            return_value=argparse.Namespace(**(Trainer.default_attributes)))
```
Comment (Borda) [inline tests/trainer/test_trainer.py]:
```suggestion
            return_value=argparse.Namespace(**Trainer.default_attributes))
```

File: tests/trainer/test_trainer.py [hunk 4]
```diff
@@ -600,3 +601,141 @@ def test_end(self, outputs):
...
+    assert not test_callback.on_test_start_called
+    assert not test_callback.on_test_end_called
+
+    trainer.test()
+
+    assert test_callback.on_test_start_called
+    assert test_callback.on_test_end_called
+
+# if __name__ == '__main__':
+#     pytest.main([__file__])
```
Comment (Borda) [inline tests/trainer/test_trainer.py]:
```suggestion
```
duplicated test

**Comment by williamFalcon** [comment]
@Borda how do we change the author to the original one?

File: pytorch_lightning/trainer/trainer.py [hunk 5]
```diff
@@ -822,6 +828,28 @@ def slurm_job_id(self) -> int:
...
+        trainer_default_params = Trainer.default_attributes()
+
+        for arg in trainer_default_params:
+            parser.add_argument('--{0}'.format(arg), default=trainer_default_params[arg], dest=arg)
+
+        return parser
+
+    @classmethod
+    def from_argparse_args(cls, args):
+
```
Comment (awaelchli) [inline pytorch_lightning/trainer/trainer.py]: shouldn't there be documentation where the blank line is? did it get lost?

Comment (Borda) [reply pytorch_lightning/trainer/trainer.py]: yeah it should...

**Pull request merged**